src/APL/saleor-cloud/saleor-cloud-apl.ts

```diff
@@ -102,25 +97,32 @@ export class SaleorCloudAPL implements APL {
       debug("Failed to reach API call:  %s", extractErrorMessage(error));
       debug("%O", error);
 
-      return undefined;
+      throw new SaleorCloudAplError("FAILED_TO_REACH_API", `${extractErrorMessage(error)}`);
     });
 
     if (!response) {
       debug("No response from the API");
-      return undefined;
+
+      throw new SaleorCloudAplError("FAILED_TO_REACH_API", "Response couldnt be resolved");
     }
 
-    try {
-      validateResponseStatus(response);
-      debug("Response status valid");
-    } catch {
-      debug("Response status not valid");
+    if (response.status >= 500) {
+      throw new SaleorCloudAplError("FAILED_TO_REACH_API", `Api responded with ${response.status}`);
+    }
+
+    if (response.status === 404) {
+      debug("No auth data for given saleorApiUrl");
       return undefined;
     }
 
     const parsedResponse = (await response.json().catch((e) => {
       debug("Failed to parse response: %s", extractErrorMessage(e));
       debug("%O", e);
+
+      throw new SaleorCloudAplError(
+        "RESPONSE_BODY_INVALID",
```

Comment:
Maybe these error codes should be an enum that we export from the library, so that it's easier for an app to handle these errors?